Add ability to granualarly filter events (#364)
jenkinsci/datadog-plugin: README.md

NOTE: `event_type` is always set to `system` for above events and metrics.

#### Security events type

| Event name                  | Triggered on                            | Default tags                                     | Associated RATE metric       |
|-----------------------------|-----------------------------------------|--------------------------------------------------|------------------------------|
| UserAuthenticated          | `SecurityListener#authenticated`        | `event_type`, `jenkins_url`, `user_id`           | `jenkins.user.authenticated` |
| UserFailedToAuthenticate | `SecurityListener#failedToAuthenticate` | `event_type`, `jenkins_url`, `user_id`           | `jenkins.user.access_denied` |
| UserLoggedOut              | `SecurityListener#loggedOut`            | `event_type`, `jenkins_url`, `user_id`           | `jenkins.user.logout`        |

NOTE: `event_type` is always set to `security` for above events and metrics.

#### Filtering events

This plugin allows for events to be filtered by the event type as well as the specific event names listed

> above. To include/exclude all events of the system or security type, in the UI, uncheck the checkboxes for these events. In a groovy script, fetch the Datadog global descriptor and call either `d.setEmitSystemEvents()` or `d.setEmitSecurityEvents()`. As listed in the [environment variables](#environment-variables) section, set the environment variables for the emitting security or system events.

Reviewer: ```suggestion
above.

To include/exclude all events of the system or security type:
1. In the UI, uncheck the checkboxes for these events.
2. In a groovy script, fetch the Datadog global descriptor and call either `d.setEmitSystemEvents()` or `d.setEmitSecurityEvents()`. 
3. As listed in the [environment variables](#environment-variables) section, set the environment variables for the emitting security or system events.
```
Might help to break up this text into step-by-step instructions.

Author: These are alternatives rather than step by step instructions, let me know if there's a better way to clarify this!

Reviewer: Ah, I see! Perhaps it would be clearer if we made it a bulleted list like this then:

```suggestion
above. To include/exclude all events of the system or security type:
- **In the UI**: Uncheck the checkboxes for these events.
- **In a groovy script**: Fetch the Datadog global descriptor and call either `d.setEmitSystemEvents()` or `d.setEmitSecurityEvents()`.
- **In the [environment variables](#environment-variables) section**: Set the environment variables for the emitting security or system events.
```

This could make it easier to scan for users.

> To get more specific control over what events are sent, three configuration options are provided to allow a comma-separated include/exclude list of strings of event names. The include/exclude list has precedence over the filtering by event type. For example, `security` events can be toggled off, but including `UserAuthenticated` takes precedence, so only `UserAuthenticated` events will be sent from the `security` type. In the UI, text boxes are provided for both the included and excluded lists. In a groovy script, the methods `d.setIncludeEvents()` and `d.setExcludeEvents()` taking in a comma-separated list of event names as input as well is another valid configuration method. Lastly, there are provided [environment variables](#environment-variables) for manually setting included/excluded lists.

Reviewer: ```suggestion
To get more specific control over what events are sent, three configuration options are provided to allow a comma-separated include/exclude list of strings of event names. The include/exclude list has precedence over filtering by event type. For example, `security` events can be toggled off, but including `UserAuthenticated` takes precedence, so only `UserAuthenticated` events will be sent from the `security` type. In the UI, text boxes are provided for both the included and excluded lists. In a groovy script, the methods `d.setIncludeEvents()` and `d.setExcludeEvents()` accept a comma-separated list of event names as input which is another valid configuration method. Lastly, there are provided [environment variables](#environment-variables) for manually setting included/excluded lists.
```
